x/sp/keeper/msg_server.go

```diff
@@ -327,13 +333,18 @@ func (k msgServer) UpdateSpStoragePrice(goCtx context.Context, msg *types.MsgUpd
 		FreeReadQuota: msg.FreeReadQuota,
 	}
 	k.SetSpStoragePrice(ctx, spStorePrice)
-	err := k.UpdateSecondarySpStorePrice(ctx)
-	if err != nil {
-		return nil, errors.Wrapf(err, "update secondary sp store price failed")
-	}
+
 	return &types.MsgUpdateSpStoragePriceResponse{}, nil
 }
 
+func IsLastDaysOfTheMonth(now time.Time, days int) bool {
+	year, month, _ := now.Date()
+	location := now.Location()
```

Comment:
is it possible that different region get different result for `now.Location()`?

Reply:
`now` is passed with UTC timezone. I refactor the code to make it clear.